src/python/dxpy/scripts/dx.py

```diff
@@ -6002,6 +6003,21 @@ def positive_integer(value):
 parser_unarchive.set_defaults(func=archive, request_mode="unarchival")
 register_parser(parser_unarchive, categories='fs')
 
+#####################################
+# extract_dataset
+#####################################
+parser_extract_dataset = subparsers.add_parser('extract_dataset', help='(Preview feature) Retrieves the data, or generates SQL to retrieve the data, from a dataset or cohort for a set of entity.fields.',
+                                   description='(Preview feature) Retrieves the data, or generates SQL to retrieve the data, from a dataset or cohort for a set of entity.fields.',
+                                   prog='dx extract_dataset')
+parser_extract_dataset.add_argument('path', help='v3.0 Dataset or Cohort object ID (project-id:record-id where ":record-id" indicates the current selected project) or name')
+parser_extract_dataset.add_argument('-ddd', '--dump-dataset-dictionary', action="store_true", default=False, help='If provided, the three dictionary files: <record_name>.data_dictionary.csv, <record_name>.entity_dictionary.csv, <record_name>.codings.csv are generated using the -delimiter specified, if applicable, in the output directory, if not provided, in the current location')
+parser_extract_dataset.add_argument('--fields', nargs='+', help='A comma-separated STRING  where each value is the phenotypic entity name and field name separated by a dot. If multiple entities are provided, they are inner-joined into one return. For example: <entity_name>.<field_name>,<entity_name>.<field_name>')
```

Comment:
Is this help string from the spec? Not sure why STRING is capitalized. @amesma0 @andydnanexus could one of you please review all the help strings from line 6009-6017?

Reply:
Confused on "Retrieves the data, or generates SQL".

Reply:
@arunbhat - yes, help strings come from Spec.
@amesma0 - if only --fields in passed in input, data is retrieved and returned. If --fields and --sql both are passed, a SQL to retrieve the data is generated and returned.

Reply:
Can you just add that part in?

Reply:
Updated!